src/server/game/Maps/Map.cpp

```diff
@@ -879,10 +879,15 @@ void Map::RemovePlayerFromMap(Player* player, bool remove)
 {
     sScriptMgr->OnPlayerLeaveMap(this, player);
 
+    player->getHostileRefManager().deleteReferences(); // multithreading crashfix
```

Comment:
Kinda weird that only this is cleaned.
What about the references on the attacker, and the attack state and all the other related things?